### PR TITLE
Lint by extension

### PR DESCRIPTION
**Pull request by nelsonpecora**
* allows linting components, pages, and public urls by extension
* note: this changes the public url linting to _also_ check the `.html` rendered page
* note: this still checks recursively, so children's templates will also be linted
* note: you may lint against _any_ extension, not just `.html`, e.g. `domain.com/_components/my-cool-feed.rss`

screen shot 2018-07-05 at 12 09 43 pm

### CODE REVIEW

File: lib/cmd/lint.js [hunk 1]
```diff
@@ -121,6 +154,13 @@ function checkComponent(url, prefix, concurrency) {
  * @return {Stream}
  */
 function checkPage(url, prefix, concurrency) {
+  let ext = '';
+
+  // if we're checking a page with an extension, cut it off from the url and pass it into the component checking
```
Comment (reubenson) [inline lib/cmd/lint.js]: i don' think clay supports non-html renderers for pages? might just update the comment to read `if we're checking a page with a .html extension [...]`?

Comment (jonwinton) [reply lib/cmd/lint.js]: It can support any type of renderer, it's just a question of if one has been built yet.

Comment (nelsonpecora) [reply lib/cmd/lint.js]: Clay supports a _lot_ of different renderers!

screen shot 2018-07-05 at 1 00 17 pm

Comment (reubenson) [reply lib/cmd/lint.js]: oh! 👌

File: lib/cmd/lint.js [hunk 2]
```diff
@@ -156,7 +222,7 @@ function checkPublicUrl(url, concurrency) {
     .flatMap(({ uri, prefix }) => {
       const pageURL = prefixes.uriToUrl(prefix, uri);
 
-      return h([h.of({ type: 'success', message: url }), checkPage(pageURL, prefix, concurrency)]).merge();
+      return h([h.of({ type: 'success', message: url }), checkPage(`${pageURL}.html`, prefix, concurrency)]).merge();
```
Comment (reubenson) [inline lib/cmd/lint.js]: should pass `${ext}` here instead of hard-coding `.html`?

Comment (nelsonpecora) [reply lib/cmd/lint.js]: Nope! Public urls always point to page html

Comment (reubenson) [reply lib/cmd/lint.js]: erm, right then!

**Pull request merged**